AppSyncRealTimeClient/ConnectionProvider/AppsyncRealtimeConnection/RealtimeConnectionProviderResponse.swift

```diff
@@ -116,4 +145,32 @@ extension RealtimeConnectionProviderResponse {
 
         return false
     }
+
+    func isUnauthorizationError() -> Bool {
+        // It is expected to contain payload with corresponding error information
+        guard let payload = payload,
+              let errors = payload["errors"],
+              case let .array(errorsArray) = errors else {
+            return false
+        }
+
+        // The observed response from the service
+        // { "payload": {
+        //      "errors": [{
+        //          "errorType":"com.amazonaws.deepdish.graphql.auth#UnauthorizedException",
+        //          "message":"You are not authorized to make this call.",
+        //          "errorCode":400 }]},
+        //  "type":"connection_error" }
+        return errorsArray.contains { error in
+            guard case let .object(errorObject) = error else {
+                return false
+            }
+
+            if errorObject["errorType"] == "com.amazonaws.deepdish.graphql.auth#UnauthorizedException" {
+                return true
+            }
```

Comment:
should we also check for the `errorCode` value? and to make the check more robust I'd check if the suffix of the `errorType` value is `UnauthorizedException`

Reply:
I'm not sure about the `errorCode` because if we check for it by itself then we may incorrect categorize 400 as Unauthorized in this code, since I'm not sure when errorCode is returned. If we do both Unauthorized && 400 then if for some reason AppSync service returns a different errorCode, then this code will break. I think just making the errorType check more robust is the ideal state right now. I updated to do a `.contains` check instead of an equality check